File: src/main/kotlin/com/example/testfordatabase/user/UserRepository.java
package com.example.testfordatabase.user;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
@Repository
public interface UserRepository extends CrudRepository<User, Long> {
    Optional<User> findByEmail(String username);

    Optional<User> findByUsername(String username);
}

File: src/main/kotlin/com/example/testfordatabase/user/UserRepository.kt
package com.example.testfordatabase.user

import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<MyUser?, Long?> {
    fun findByEmail(username: String?): MyUser?
    fun findByUsername(username: String?): MyUser?
}